From a397b94ffb5cf201aef4fdd9ccd2b28780713bed Mon Sep 17 00:00:00 2001
From: David Engel <dengel@mythtv.org>
Date: Tue, 11 Nov 2014 16:13:07 -0600
Subject: [PATCH] Properly identify channel-specific rules in Recording Rules.

This was broken when channel rules were converted to all rules using
the this channel filter.  Based on a patch from lomion@aon.at.

Fixes #12267
---
 .../programs/mythfrontend/programrecpriority.cpp   |   63 +++++++-------------
 1 file changed, 21 insertions(+), 42 deletions(-)

diff --git a/mythtv/programs/mythfrontend/programrecpriority.cpp b/mythtv/programs/mythfrontend/programrecpriority.cpp
index 846f750..01a61e3 100644
--- a/mythtv/programs/mythfrontend/programrecpriority.cpp
+++ b/mythtv/programs/mythfrontend/programrecpriority.cpp
@@ -1418,25 +1418,17 @@ void ProgramRecPriority::UpdateList()
         item->SetText(tempTime, "lastrecordedtime", state);
 
         QString channame = progInfo->channame;
-        if ((progInfo->recType == kAllRecord) ||
-            (progInfo->recType == kOneRecord) ||
-            (progInfo->recType == kDailyRecord) ||
-            (progInfo->recType == kWeeklyRecord))
-            channame = tr("Any");
-        item->SetText(channame, "channel", state);
         QString channum = progInfo->chanstr;
-        if ((progInfo->recType == kAllRecord) ||
-            (progInfo->recType == kOneRecord) ||
-            (progInfo->recType == kDailyRecord) ||
-            (progInfo->recType == kWeeklyRecord))
-            channum = tr("Any");
-        item->SetText(channum, "channum", state);
         QString callsign = progInfo->chansign;
-        if ((progInfo->recType == kAllRecord) ||
-            (progInfo->recType == kOneRecord) ||
-            (progInfo->recType == kDailyRecord) ||
-            (progInfo->recType == kWeeklyRecord))
+        if (!(progInfo->GetRecordingRule()->m_filter & 1024) &&
+            progInfo->GetRecordingRule()->m_searchType != kManualSearch)
+        {
+            channame = tr("Any");
+            channum = tr("Any");
             callsign = tr("Any");
+        }
+        item->SetText(channame, "channel", state);
+        item->SetText(channum, "channum", state);
         item->SetText(callsign, "callsign", state);
 
         QString profile = progInfo->profile;
@@ -1535,37 +1527,24 @@ void ProgramRecPriority::updateInfo(MythUIButtonListItem *item)
         m_lastRecordedTimeText->SetText(tempTime);
     }
 
-    if (m_channameText)
+    if (m_channameText || m_channumText || m_callsignText)
     {
         QString channame = pgRecInfo->channame;
-        if ((pgRecInfo->rectype == kAllRecord) ||
-            (pgRecInfo->rectype == kOneRecord) ||
-            (pgRecInfo->rectype == kDailyRecord) ||
-            (pgRecInfo->rectype == kWeeklyRecord))
-            channame = tr("Any");
-        m_channameText->SetText(channame);
-    }
-
-    if (m_channumText)
-    {
         QString channum = pgRecInfo->chanstr;
-        if ((pgRecInfo->rectype == kAllRecord) ||
-            (pgRecInfo->rectype == kOneRecord) ||
-            (pgRecInfo->rectype == kDailyRecord) ||
-            (pgRecInfo->rectype == kWeeklyRecord))
-            channum = tr("Any");
-        m_channumText->SetText(channum);
-    }
-
-    if (m_callsignText)
-    {
         QString callsign = pgRecInfo->chansign;
-        if ((pgRecInfo->rectype == kAllRecord) ||
-            (pgRecInfo->rectype == kOneRecord) ||
-            (pgRecInfo->rectype == kDailyRecord) ||
-            (pgRecInfo->rectype == kWeeklyRecord))
+        if (!(pgRecInfo->GetRecordingRule()->m_filter & 1024) &&
+            pgRecInfo->GetRecordingRule()->m_searchType != kManualSearch)
+        {
+            channame = tr("Any");
+            channum = tr("Any");
             callsign = tr("Any");
-        m_callsignText->SetText(callsign);
+        }
+        if (m_channameText)
+            m_channameText->SetText(channame);
+        if (m_channumText)
+            m_channumText->SetText(channum);
+        if (m_callsignText)
+            m_callsignText->SetText(callsign);
     }
 
     if (m_recProfileText)
-- 
1.7.10.2

